### PR TITLE
feat: parameterize modbus configuration

### PR DESCRIPTION
**Pull request by epcmatt**
Add parameters for sunspec.core.client.SunSpecClientDevice
as optional parameters with defaults for load function.
Removes hard coded configuration and allows modbus/sunspec to be
configured during sunspec object initialization.

### CODE REVIEW

**Comment by epcmatt** [comment]
Didn't add any test for this, but probably should.

**Comment by altendky** [comment]
> Didn't add any test for this, but probably should.

This repository has tests but is woefully under-tested.  I don't particularly expect that to get corrected.  If we were to do a rewrite I would plan on it being in a new repository (or multiple really) with strict test coverage requirements etc.  For now for here, I'm not picky.

File: epyqlib/hildevice.py [hunk 1]
```diff
@@ -914,15 +914,29 @@ class SunSpecDevice:
...
+        slave_id=1,
+        device_type=sunspec.core.client.RTU,
+        name='/dev/ttyUSB0',
+        baudrate=115200,
+        timeout=1,
+        parity=sunspec.core.client.PARITY_NONE,
+        ipaddr=None,
+        ipport=502,
+        pathlist=None,
+        trace=False):
```
Comment (altendky) [inline epyqlib/hildevice.py:927]:
```suggestion
        trace=False,
    ):
```

File: epyqlib/hildevice.py [hunk 2]
```diff
@@ -914,15 +914,29 @@ class SunSpecDevice:
...
+                slave_id=slave_id,
+                device_type=device_type,
+                name=name,
+                baudrate=baudrate,
+                timeout=timeout,
+                parity=parity,
+                ipaddr=ipaddr,
+                ipport=ipport,
+                pathlist=pathlist,
+                trace=trace)
```
Comment (altendky) [inline epyqlib/hildevice.py:939]:
```suggestion
                trace=trace,
            )
```

Comment (epcmatt) [reply epyqlib/hildevice.py]: Updated in second commit.

File: epyqlib/hildevice.py (hunk 1 above)
Comment (epcmatt) [reply epyqlib/hildevice.py]: Updated in second commit.

**Comment by epcmatt** [comment]
Added second commit with additional load methods.

File: epyqlib/hildevice.py [hunk 3]
```diff
@@ -914,14 +915,65 @@ class SunSpecDevice:
...
+                name=name,
+                baudrate=baudrate,
+                timeout=timeout,
+                parity=parity,
+            )
+
+    def load_tcp(self,
+        slave_id=1,
+        timeout=1,
+        ipaddr=None,
```
[the remark this answers is not in the document]
Comment (epcmatt) [reply epyqlib/hildevice.py]: agreed, made update

File: epyqlib/hildevice.py [hunk 4]
```diff
@@ -914,14 +915,65 @@ class SunSpecDevice:
...
+        ipport=502,
```
[the remark this answers is not in the document]
Comment (epcmatt) [reply epyqlib/hildevice.py]: agreed, made update

File: epyqlib/hildevice.py [hunk 5]
```diff
@@ -914,14 +915,65 @@ class SunSpecDevice:
...
+        timeout=1,
+        parity=sunspec.core.client.PARITY_NONE,
+        ipaddr=None,
+        ipport=502,
+        pathlist=None,
+        trace=False,
+    ):
+        warnings.warn("SunSpecDevice.load() method is deprecated and will be " +
+                        "removed in the future. Use SunSpecDevice.load_rtu() " + 
+                        "or SunSpecDevice.load_tcp() instead.")
```
[the remark this answers is not in the document]
Comment (epcmatt) [reply epyqlib/hildevice.py]: Left original .load() method for now. Black works for me but I'm not comfortable running it on the whole repository myself.

**Comment by epcmatt** [comment]
Updates committed to fork

File: epyqlib/hildevice.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (altendky) [reply epyqlib/hildevice.py]: We'll black after this is in and also the the files tab PR.

**Pull request merged**